### PR TITLE
Bus net handling in Yosys JSON to netlist conversion

### PR DESCRIPTION
**Pull request by mkurc-ant**
When using the `splitnets` pass Yosys preserves `offset` of each resulting net while it appends the index `[<index>]` part to its name. This confuses the conversion script and results in index duplication as it considers net literaly named eg. `cnt[3]` as 3rd bit of the bus `cnt` and stores it as `cnt[3][3]`.

This PR addresses this unwanted behavior by not considering 1-bit nets nets named `<name>[<index>]` with offset==index as part of a bus.

### CODE REVIEW

File: fpga_interchange/yosys_json.py [hunk 1]
```diff
@@ -18,20 +18,44 @@
...
     True
+    >>> is_bus("net[3]", [100], offset=3, upto=0)
+    False
 
     """
+
+    # In case of a port which is a result of the "splitnets" pass the
+    # original offset is preserved even though the port is no longer a bus.
+    #
+    # Detect it here by looking at the bit indes in brackets (which is a part
```
Comment (acomodi) [inline fpga_interchange/yosys_json.py:43]: typo: indes -> index
```suggestion
    # Detect it here by looking at the bit index in brackets (which is a part
```

File: fpga_interchange/yosys_json.py [hunk 2]
```diff
@@ -18,20 +18,44 @@
...
+    >>> is_bus("net", [2, 3], offset=0, upto=0)
     True
-    >>> is_bus([2], offset=1, upto=0)
+    >>> is_bus("net", [2], offset=1, upto=0)
     True
-    >>> is_bus([2], offset=0, upto=1)
+    >>> is_bus("net", [2], offset=0, upto=1)
     True
+    >>> is_bus("net[3]", [100], offset=3, upto=0)
+    False
```
Comment (acomodi) [inline fpga_interchange/yosys_json.py:37]: Should probably add a test for the following case:
```
is_bus("net[3]", [100], offset=4, upto=0)
True
```

As far as I understand from the below comment, if the offset and index in the brackets are not the same, the net should be considered as a bus

**Review by acomodi — COMMENTED** [review]
Comments in

File: fpga_interchange/yosys_json.py (hunk 1 above)
Comment (mkurc-ant) [reply fpga_interchange/yosys_json.py]: fixed.

File: fpga_interchange/yosys_json.py (hunk 2 above)
Comment (mkurc-ant) [reply fpga_interchange/yosys_json.py]: I added the test case.

**Review by acomodi — APPROVED** [review]
LGTM

**Pull request merged**